runtime-watcher/internal/cacertificatehandler/ca_certificate_handler_test.go

```diff
@@ -0,0 +1,131 @@
+package cacertificatehandler_test
+
+import (
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/kyma-project/runtime-watcher/skr/internal/cacertificatehandler"
+	"github.com/kyma-project/runtime-watcher/skr/internal/tlstest"
+)
+
+func TestGetCertificatePool1(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name             string
+		certificateCount int
+		certPath         string
+	}{
+		{
+			name:             "certificate pool with one certificate",
+			certificateCount: 1,
+			certPath:         "ca-1.cert",
+		},
+		{
+			name:             "certificate pool with two certificates",
+			certificateCount: 2,
+			certPath:         "ca-2.cert",
+		},
+	}
+	for _, tt := range tests {
+		testCase := tt
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			file, err := os.CreateTemp("", testCase.certPath)
+			require.NoError(t, err)
+
+			err = writeCertificatesToFile(file, testCase.certificateCount)
+			require.NoError(t, err)
+
```

Comment:
From that point you no longer need the os.File abstraction, you only need it's name. Please close the file here to ensure all os-level buffers are flushed and any file-related operations actually see the written data.